config/config-mp/src/main/java/io/helidon/config/mp/MpConfigBuilder.java

```diff
@@ -0,0 +1,347 @@
+        }
+        return this;
+    }
+
+    @Override
+    public Config build() {
+        if (useDefaultSources) {
+            sources.add(new OrdinalSource(MpConfigSources.systemProperties(), 400));
+            sources.add(new OrdinalSource(MpConfigSources.environmentVariables(), 300));
+            // microprofile-config.properties
+            MpConfigSources.classPath(classLoader, "META-INF/microprofile-config.properties")
+                    .stream()
+                    .map(OrdinalSource::new)
+                    .forEach(sources::add);
+        }
+        // built-in converters - required by specification
+        addBuiltIn(converters, Boolean.class, ConfigMappers::toBoolean);
+        addBuiltIn(converters, Boolean.TYPE, ConfigMappers::toBoolean);
+        addBuiltIn(converters, Byte.class, Byte::parseByte);
+        addBuiltIn(converters, Byte.TYPE, Byte::parseByte);
+        addBuiltIn(converters, Short.class, Short::parseShort);
+        addBuiltIn(converters, Short.TYPE, Short::parseShort);
+        addBuiltIn(converters, Integer.class, Integer::parseInt);
+        addBuiltIn(converters, Integer.TYPE, Integer::parseInt);
+        addBuiltIn(converters, Long.class, Long::parseLong);
+        addBuiltIn(converters, Long.TYPE, Long::parseLong);
+        addBuiltIn(converters, Float.class, Float::parseFloat);
+        addBuiltIn(converters, Float.TYPE, Float::parseFloat);
+        addBuiltIn(converters, Double.class, Double::parseDouble);
+        addBuiltIn(converters, Double.TYPE, Double::parseDouble);
+        addBuiltIn(converters, Character.class, MpConfigBuilder::toChar);
+        addBuiltIn(converters, Character.TYPE, MpConfigBuilder::toChar);
+        addBuiltIn(converters, Class.class, MpConfigBuilder::toClass);
+
+        // built-in converters - Helidon
+        //javax.math
+        addBuiltIn(converters, BigDecimal.class, ConfigMappers::toBigDecimal);
+        addBuiltIn(converters, BigInteger.class, ConfigMappers::toBigInteger);
+        //java.time
+        addBuiltIn(converters, Duration.class, ConfigMappers::toDuration);
+        addBuiltIn(converters, Period.class, ConfigMappers::toPeriod);
+        addBuiltIn(converters, LocalDate.class, ConfigMappers::toLocalDate);
+        addBuiltIn(converters, LocalDateTime.class, ConfigMappers::toLocalDateTime);
+        addBuiltIn(converters, LocalTime.class, ConfigMappers::toLocalTime);
+        addBuiltIn(converters, ZonedDateTime.class, ConfigMappers::toZonedDateTime);
+        addBuiltIn(converters, ZoneId.class, ConfigMappers::toZoneId);
+        addBuiltIn(converters, ZoneOffset.class, ConfigMappers::toZoneOffset);
+        addBuiltIn(converters, Instant.class, ConfigMappers::toInstant);
+        addBuiltIn(converters, OffsetTime.class, ConfigMappers::toOffsetTime);
+        addBuiltIn(converters, OffsetDateTime.class, ConfigMappers::toOffsetDateTime);
+        addBuiltIn(converters, YearMonth.class, YearMonth::parse); ;
```

Comment:
Extra semicolon at the end is causing checkstyle to fail.

Reply:
fixed